packages/dds/tree/src/tableSchema.ts

```diff
@@ -409,12 +417,14 @@ export namespace System_TableSchema {
 
 				const columnId = typeof columnOrId === "string" ? columnOrId : columnOrId.id;
 
-				const cell: CellValueType | undefined = this.cells.get(columnId);
+				const cell: CellValueType | undefined = this.cells[columnId];
 				if (cell === undefined) {
 					return undefined;
 				}
 
-				this.cells.delete(columnId);
+				// eslint-disable-next-line @typescript-eslint/no-dynamic-delete -- The record's values are non-optional, so setting `undefined` as a means to remove the cell is not supported.
```

Comment:
Our linter doesn't like using objects as maps. Personally I agree with it.

The link says https://typescript-eslint.io/rules/no-dynamic-delete/ "Consider using a Map or Set if you’re using an object as a key-value collection."

Have you considered using a Map node instead of a Record?

Given our own selection of linter rules, I think libraries we offer should probably avoid record nodes unless there is a particularly good reason to use them.

If we have such a reason, it should probably be documented in the schema where the choice to use record is made (which currently has no docs)